core/src/test/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceServiceSpec.scala

```diff
@@ -597,30 +708,38 @@ class MultiCloudWorkspaceServiceSpec extends AnyFlatSpec with Matchers with Opti
                 Map.empty
               )
             )
-
             _ = clone.toWorkspaceName shouldBe cloneName
             _ = clone.workspaceType shouldBe McWorkspace
 
-//            To be added back with WOR-697
-//            jobs <- slickDataSource.inTransaction { access =>
-//              for {
-//                // the newly cloned workspace should be persisted
-//                clone_ <- access.workspaceQuery.findByName(cloneName)
-//                _ = clone_.value.workspaceId shouldBe clone.workspaceId
-//
-//                // a new resource monitor job should be created
-//                jobs <- access.WorkspaceManagerResourceMonitorRecordQuery
-//                  .selectByWorkspaceId(clone.workspaceIdAsUUID)
-//              } yield jobs
-            //           }
-          } yield clone.completedCloneWorkspaceFileTransfer shouldBe None
-//            jobs.size shouldBe 1
-//            jobs.head.jobType shouldBe JobType.CloneWorkspaceResult
-//            jobs.head.workspaceId.value.toString shouldBe clone.workspaceId
-          ,
+            jobs <- slickDataSource.inTransaction { access =>
+              for {
+                // the newly cloned workspace should be persisted
+                clone_ <- access.workspaceQuery.findByName(cloneName)
+                _ = clone_.value.workspaceId shouldBe clone.workspaceId
+
+                // a new resource monitor job should be created
+                jobs <- access.WorkspaceManagerResourceMonitorRecordQuery
+                  .selectByWorkspaceId(clone.workspaceIdAsUUID)
+              } yield jobs
+            }
+          } yield {
+            clone.completedCloneWorkspaceFileTransfer shouldBe None
+            jobs.size shouldBe 1
+            jobs.head.jobType shouldBe JobType.CloneWorkspaceResult
+            jobs.head.workspaceId.value.toString shouldBe clone.workspaceId
+            clone.workspaceIdAsUUID
+          },
           Duration.Inf
         )
+        verify(mcWorkspaceService.workspaceManagerDAO, times(1))
+          .cloneAzureStorageContainer(
+            equalTo(testData.azureWorkspace.workspaceIdAsUUID),
+            equalTo(cloneWorkspaceUUID),
+            equalTo(sourceContainerUUID),
+            equalTo(getStorageContainerName(cloneWorkspaceUUID)),
+            equalTo(CloningInstructionsEnum.RESOURCE),
+            any()
+          ) shouldBe null // expects an assertion to be returned
```

Comment:
Is there a better way to get this to compile? I don't really understand why it is necessary here, as we are able to do mock assertions at the end of other test cases without explicitly returning an assertion.

Reply:
It's because you're running it inside `withMockedMultiCloudWorkspaceService`, which has takes a function with the signature `MultiCloudWorkspaceService => Assertion`: so it expects an assertion to be returned at the end.

Reply:
You could move the verify up into the yield block and as long as it's not the last line in the yield it would work. You'd also have to stop assigning the result to `cloneWorkspaceUUID`

Reply:
Thanks, much better!